source/extensions/filters/http/cache/key.proto

```diff
@@ -9,8 +9,14 @@ message Key {
   string path = 3;
   string query = 4;
   // True for http://, false for https://.
-  bool clear_http = 5;
+  bool clear_http = 5 [deprecated = true]; // Use scheme instead.
```

Comment:
I think it would be reasonable to give it a month or two.

Reply:
I don't think we should deprecate until the new one actually works?  Also should we mark the new one as // [#not-implemented-hide:] to indicate it is not yet implemented?  Generally I encourage API changes to land with the code.  WDYT about delaying this until you've got the code prepared?

Reply:
Oh, good point! I thought clear_http wasn't used yet but there is one file where it's used (http_cache.cc). I've made the change there. I'll also see if I can add test coverage (I'll need another commit in this PR).

This brings up the question of whether we need to protect this change more. Since this PR also changes the code that creates the cache key, then it could cause a cache flush for anyone who's using the CacheFilter already. The CacheFilter is still in Alpha, so maybe it's ok? The one example committed to the repo (SimpleHttpCache) is based on an in-memory cache which gets flushed on restart anyway.

Reply:
Added tests. Note that the UNSPECIFIED option isn't used currently. It will be used once we refactor LookupRequest::LookupRequest to call CachePolicy::createCacheKey to create the cache key. Then the cache policy can specify whether we want to include the scheme or not.

We've already made such a change internally and will work on upstreaming it in the coming weeks, along with many more changes, so I could wait until then and get this PR along with the rest of the relevant changes. However, I thought it would be a good idea to get this one in ASAP since it changes the structure of cache keys and will be more tricky to upstream the longer we wait.

And sorry for the messy PR, I'm still getting used to the Envoy Github workflow as a Googler...

Reply:
yeah I think with the code changes added we can in-place deprecate as long as we have sign-off from one of the cache owners.  I'll add them in to the review